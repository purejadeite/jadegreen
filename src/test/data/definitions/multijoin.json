{
	"id": "book",
	"sheets": [
		{
			"comment": "結合先シート",
			"id": "sub",
			"target": {
			    "cell": {
			    	"row": 3,
			    	"column": 3,
			    	"value": "validation"
			    }
			},
		    "cells": [
		        {"id": "logicalClassName", "row": 1, "column": 3},
		        {"id": "className", "row": 2, "column": 3},
		        {"id": "fields", "row": 6, "options": [{"type": "group", "groupId": "fieldName", "valueId": "validators"}],
		            "columns": [
		                {"id": "fieldName", "column": 3},
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "validationType", "column": 4, "options": [{"type": "mapping", "map": {"notnull": "NotNull"}, "lazy": true}]},
		                {"id": "arg0", "column": 5, "options": [{"type": "tonull"}]},
		                {"id": "arg1", "column": 6, "options": [{"type": "tonull"}]},
		                {"id": "arg2", "column": 7, "options": [{"type": "tonull"}]},
		                {"id": "arg3", "column": 8, "options": [{"type": "tonull"}]}
		            ]
		        }
		    ]
		},
		{
			"comment": "結合あり1件読み込み",
			"id": "main",
			"output": true,
			"target": {
			    "cell": {
			    	"row": 5,
			    	"column": 3,
			    	"value": "bean"
			    }
			},
		    "cells": [
		        {"id": "logicalClassName", "row": 1, "column": 3},
		        {"id": "className", "row": 2, "column": 3},
		        {"id": "package", "row": 3, "column": 3},
		        {"id": "imports", "row": 4, "column": 3, "splitter": "\n"},
		        {"id": "fields", "row": 8, "options": [{"type": "sort", "keyId": "fieldName"}],
		            "columns": [
		                {"id": "fieldName", "column": 3},
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "method", "column": 3, "options": [{"type": "uppercamel"}]},
		                {"id": "static", "column": 3, "options": [{"type": "upperunderscore"}]},
		                {"id": "type", "column": 4},
		                {"id": "initial", "column": 5},
		                {"id": "validation", "join": {"myKeyId": "className", "mytableKeyId": "fieldName", "sheetId": "sub", "keyId": "className", "tableKeyId": "fieldName", "valueId": "validators"}}
		            ]
		        },
		        {"id": "logicalClassNameValid", "join": {"myKeyId": "className", "sheetId": "sub", "keyId": "className", "valueId": "logicalClassName"}}
		    ]
		}
	]
}