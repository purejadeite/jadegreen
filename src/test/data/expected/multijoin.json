[
	{
		"sheetName": "bean-sample1",
		"logicalClassName": "シート間の結合",
		"logicalClassNameValid": "シート間の結合(結合先)",
		"className": "Sample1",
		"package": "com.purejadeite.generated",
		"imports": [
			"java.io.Serializable",
			"java.util.List",
			"java.util.ArrayList",
			"com.purejadeite.generated.LineBean"
        ],
		"fields": [
			{
				"logicalFieldName": "ファイル有無フラグ",
				"fieldName": "fileExistFlag1",
				"method": "FileExistFlag1",
				"static": "FILE_EXIST_FLAG1",
				"type": "Boolean",
				"initial": "Boolean.TRUE",
				"validation": [
		   			{
						"logicalFieldName": "ファイル有無フラグ",
						"fieldName": "fileExistFlag1",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "ファイル名",
				"fieldName": "fileName",
				"method": "FileName",
				"static": "FILE_NAME",
				"type": "String",
				"initial": "null",
				"validation": [
		   			{
		   				"logicalFieldName": "ファイル名",
						"fieldName": "fileName",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "ファイル名日本語",
				"fieldName": "fileNameJp",
				"method": "FileNameJp",
				"static": "FILE_NAME_JP",
				"type": "String",
				"initial": "null",
				"validation": null
			},
			{
				"logicalFieldName": "ファイルパス",
				"fieldName": "filePath",
				"method": "FilePath",
				"static": "FILE_PATH",
				"type": "String",
				"initial": null,
				"validation": [
					{
						"logicalFieldName": "ファイルパス",
						"fieldName": "filePath",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
		   			{
						"logicalFieldName": "ファイルパス",
						"fieldName": "filePath",
						"validationType": "length",
						"arg0": "2",
						"arg1": "10",
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "行情報",
				"fieldName": "lines",
				"method": "Lines",
				"static": "LINES",
				"type": "List<LineBean>",
				"initial": "new ArrayList<>()",
				"validation": [
		   			{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
					{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "maxlength",
						"arg0": "100",
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
					{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "kana",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			}
		]
	},
	{
		"sheetName": "bean-sample2",
		"logicalClassName": "シート間の結合",
		"logicalClassNameValid": "シート間の結合(結合先)",
		"className": "Sample2",
		"package": "com.purejadeite.generated",
		"imports": [
			"java.io.Serializable",
			"java.util.List",
			"java.util.ArrayList",
			"com.purejadeite.generated.LineBean"
        ],
		"fields": [
			{
				"logicalFieldName": "ファイル有無フラグ",
				"fieldName": "fileExistFlag2",
				"method": "FileExistFlag2",
				"static": "FILE_EXIST_FLAG2",
				"type": "Boolean",
				"initial": "Boolean.TRUE",
				"validation": [
		   			{
						"logicalFieldName": "ファイル有無フラグ",
						"fieldName": "fileExistFlag2",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "ファイル名",
				"fieldName": "fileName",
				"method": "FileName",
				"static": "FILE_NAME",
				"type": "String",
				"initial": "null",
				"validation": [
		   			{
		   				"logicalFieldName": "ファイル名",
						"fieldName": "fileName",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "ファイル名日本語",
				"fieldName": "fileNameJp",
				"method": "FileNameJp",
				"static": "FILE_NAME_JP",
				"type": "String",
				"initial": "null",
				"validation": null
			},
			{
				"logicalFieldName": "ファイルパス",
				"fieldName": "filePath",
				"method": "FilePath",
				"static": "FILE_PATH",
				"type": "String",
				"initial": null,
				"validation": [
					{
						"logicalFieldName": "ファイルパス",
						"fieldName": "filePath",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
		   			{
						"logicalFieldName": "ファイルパス",
						"fieldName": "filePath",
						"validationType": "length",
						"arg0": "2",
						"arg1": "10",
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "行情報",
				"fieldName": "lines",
				"method": "Lines",
				"static": "LINES",
				"type": "List<LineBean>",
				"initial": "new ArrayList<>()",
				"validation": [
		   			{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
					{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "maxlength",
						"arg0": "100",
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
					{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "kana",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			}
		]
	},
	{
		"sheetName": "bean-sample3",
		"logicalClassName": "シート間の結合",
		"logicalClassNameValid": null,
		"className": "Sample3",
		"package": "com.purejadeite.generated",
		"imports": [
			"java.io.Serializable",
			"java.util.List",
			"java.util.ArrayList",
			"com.purejadeite.generated.LineBean"
        ],
		"fields": [
			{
				"logicalFieldName": "ファイル有無フラグ",
				"fieldName": "fileExistFlag3",
				"method": "FileExistFlag3",
				"static": "FILE_EXIST_FLAG3",
				"type": "Boolean",
				"initial": "Boolean.TRUE"
			},
			{
				"logicalFieldName": "ファイル名",
				"fieldName": "fileName",
				"method": "FileName",
				"static": "FILE_NAME",
				"type": "String",
				"initial": "null"
			},
			{
				"logicalFieldName": "ファイル名日本語",
				"fieldName": "fileNameJp",
				"method": "FileNameJp",
				"static": "FILE_NAME_JP",
				"type": "String",
				"initial": "null"
			},
			{
				"logicalFieldName": "ファイルパス",
				"fieldName": "filePath",
				"method": "FilePath",
				"static": "FILE_PATH",
				"type": "String",
				"initial": null
			},
			{
				"logicalFieldName": "行情報",
				"fieldName": "lines",
				"method": "Lines",
				"static": "LINES",
				"type": "List<LineBean>",
				"initial": "new ArrayList<>()"
			}
		]
	}
]